tools/internal/autoupdate/registry.go

```diff
@@ -0,0 +1,348 @@
+package autoupdate
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/Masterminds/semver/v3"
+	"github.com/rancher/image-mirror/internal/config"
+	"io"
+	"math"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+)
+
+type Registry struct {
+	Images        []AutoupdateImageRef
+	Latest        bool   `json:",omitempty"`
+	LatestEntry   bool   `json:",omitempty"`
```

Comment:
If you want, we could remove `LatestEntry`, since nothing uses it. Also, it is a potentially misleading name - it's hard to guarantee that the first tag is the latest one, given that we are supporting a number of different registries. Your call though.